cmd/insprd/auth/jwt/auth.go

```diff
@@ -0,0 +1,86 @@
+// Package jwtauth is responsible for implementing the auth
+// methods specified in the auth folder of the inspr pkg.
+package jwtauth
+
+import (
+	"crypto/rsa"
+	"encoding/json"
+	"errors"
+	"time"
+
+	"github.com/lestrrat-go/jwx/jwa"
+	"github.com/lestrrat-go/jwx/jwt"
+	"gitlab.inspr.dev/inspr/core/pkg/auth/models"
+)
+
+// JWTauth structure containing the private key of in the service side,
+// this key is used to parse the user keys given in the requests.
+type JWTauth struct {
+	rsaKey *rsa.PrivateKey
+}
+
+// NewJWTauth takes an *rsa.PrivateKey and returns an
+// structure that implements the auth interface
+func NewJWTauth(privateKey *rsa.PrivateKey) *JWTauth {
+	return &JWTauth{
+		rsaKey: privateKey,
+	}
+}
+
+// Validade is a wrapper that checks the token of the http request and if it's
+// valid, proceeds to execute the request and if it isn't valid returns an error
+func (JA *JWTauth) Validade(token []byte) (models.Payload, []byte, error) {
```

Comment:
Did you mean to call it "Validate"?  

Also, I think it's better to change the return to *models.Payload (so the returns when there's an error are just nil, and not an empty structure)